.pre-commit-config.yaml

```
@@ -0,0 +1,41 @@
+        name: Check Workflow Permissions
+        entry: bash -c 'for f in .github/workflows/*.yml; do grep -L "permissions:" "$f" && echo "Missing permissions in $f" && exit 1 || true; done; exit 0'
+        language: system
+        files: \.github/workflows/.*\.yml$
+        pass_filenames: false
+
+      - id: actionlint
+        name: Lint GitHub Actions
+        entry: bash -c 'which actionlint >/dev/null 2>&1 && actionlint || echo "actionlint not installed, skipping"'
+        language: system
+        files: \.github/workflows/.*\.yml$
+        pass_filenames: false
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)